ci_tools/list_docs_tovalidate.py

```diff
@@ -48,11 +48,14 @@
         '''
         if isinstance(node, (FunctionDef, ClassDef)):
             if any((node.lineno <= x <= node.end_lineno
-                    for x in line_nos)):
+                    for x in line_nos)) and node.name != 'inner_function':
                 objects.append('.'.join([prefix, node.name]))
             for child in node.body:
                 if isinstance(child, (FunctionDef, ClassDef)):
-                    child.name = '.'.join([node.name, child.name])
+                    if isinstance(node, ClassDef):
+                        child.name = '.'.join([node.name, child.name])
+                    else:
+                        child.name = 'inner_function'
```

Comment:
I'm not a huge fan of this solution as it will run faster as:
```
if isinstance(node, ClassDef):
    for child in node.body:
           if isinstance(child, (FunctionDef, ClassDef)):
                child.name = '.'.join([node.name, child.name])
else:
    for child in node.body:
           if isinstance(child, (FunctionDef, ClassDef)):
                child.name = 'inner_function'
```
but then this feels like code duplication. Maybe:
```
if isinstance(node, (FunctionDef, ClassDef)):
            if any((node.lineno <= x <= node.end_lineno
                    for x in line_nos)) and not node.name.startswith('inner_function.'):
                objects.append('.'.join([prefix, node.name]))
            prefix = node.name if isinstance(node, ClassDef) else 'inner_function'
            for child in node.body:
                if isinstance(child, (FunctionDef, ClassDef)):
                    child.name = '.'.join([prefix, child.name])
```
which also avoids evaluating the property multiple times?

Reply:
> 
Perfect